arrows/tests/test_resection_camera.h

```diff
@@ -0,0 +1,111 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include <vital/math_constants.h>
+#include <arrows/mvg/projected_track_set.h>
+#include <test_scene.h>
+#include <test_eigen.h>
+
+using namespace kwiver::vital;
+using namespace kwiver::arrows::mvg;
+using namespace std;
+
+// ----------------------------------------------------------------------------
+// Test camera resection with ideal points.
+TEST(resection_camera, ideal_points)
+{
+  // landmarks at random locations
+  landmark_map_sptr landmarks = kwiver::testing::init_landmarks(128);
+  landmarks = kwiver::testing::noisy_landmarks(landmarks, 1.0);
+
+  // camera sequence (elliptical path)
+  camera_map_sptr cameras = kwiver::testing::camera_seq();
+
+  // tracks from projections
+  track_set_sptr tracks = projected_tracks(landmarks, cameras);
+
+  const frame_id_t frmID = 0;
+
+  // corresponding image points
+  vector<vector_2d> pts_projs;
+  vector<vector_3d> pts_3d;
+  for (auto const& track : tracks->tracks())
+  {
+    auto lm_id = track->id();
+    auto lm_it = landmarks->landmarks().find(lm_id);
+    pts_3d.push_back(lm_it->second->loc());
+    auto const fts = dynamic_pointer_cast<feature_track_state>(*track->find(frmID));
+    pts_projs.push_back(fts->feature->loc());
+  }
+
+  // camera pose from the points and their projections
+  resection_camera res_cam;
+  auto cams = cameras->cameras();
+  auto cam = dynamic_pointer_cast<camera_perspective>(cams[frmID]);
+  vector<bool> inliers;
+  auto est_cam = res_cam.resection(pts_projs, pts_3d, inliers, cam->intrinsics());
```

Comment:
We should probably have a test that calls the other API that works directly on tracks and landmarks to make sure that works as well.

Also we probably want a test that doesn't use the correct intrinsics as input to make sure we can recover the intrinsics as well.

Reply:
we currently have two tests, one calling resection with points, while another one with tracks; I am working on a test with null intrinsics

Reply:
I changed the interface for the convenience method, requesting the image dimensions instead the initial calibration structure.

Reply:
added one more virtual resection(frmID, landmarks, tracks, cam.intrinsics), and a test of that

Reply:
Maybe we should provide both? (Oh, look, we *do*.) I had similar thoughts ("why are we not testing conversion via the convenience method").